Automattic/wp-calypso
# Site Logs: allow downloading logs

Closes https://github.com/Automattic/dotcom-forge/issues/1998

## Proposed Changes

* Show a download button that can download the currently selected logs locally
* Show download progress to the right side of the button


![image](https://user-images.githubusercontent.com/26530524/228360793-3fc23c51-aba0-42f1-ade9-9a037b4d64cc.png)

![image](https://user-images.githubusercontent.com/26530524/228360699-fe20ca7d-ed95-4d75-ade9-46bb51a47d2f.png)


## Testing Instructions



* Checkout this branch
* Navigate to `/site-logs/:your-atomic-site`
* Click "Download"
* Confirm a log file is downloaded for the current tab, e.g. PHP
* Confirm the log contents match the UI contents. 

## Pre-merge Checklist
